src/server/svr_jobfunc.c

```diff
@@ -4603,35 +4603,41 @@ start_end_dur_wall(void *pobj, int objtype)
 			return (-1);
 	}
 	else {
-		swcode = 3;
+		if (presv->ri_alter_flags & RESV_DURATION_MODIFIED) 
+				swcode += 4;
+		if (presv->ri_alter_flags & RESV_END_TIME_MODIFIED)
+				swcode += 2; /*calcualte start time*/
+		if (presv->ri_alter_flags & RESV_START_TIME_MODIFIED)
+				swcode += 1; /*calculate end time*/;
 	}
 
 	atemp.at_flags = ATR_VFLAG_SET;
 	atemp.at_type = ATR_TYPE_LONG;
 	switch (swcode) {
+		case  1:
+		case  2:
```

Comment:
Should I do something like

_if (swcode > 0 && swcode < 3)
     swcode = 3_

Because if state is RESV_BEING_ALTERED case 1, 2, 3 are the same.

Reply:
I would do
```C
if (flags == RESV_START_TIME_MODIFIED || flags == RESV_END_TIME_MODIFIED)
    swcode = 3;
```

Reply:
That will not work because in addition start_time and end_time,  duration could be modified too. How about something like this
```
  swcode = 3;
  if (presv->ri_alter_flags & RESV_DURATION_MODIFIED) {
	swcode +=1;
	if (presv->ri_alter_flags & RESV_END_TIME_MODIFIED)
		swcode += 2; /*calcualte start time*/
	if (presv->ri_alter_flags & RESV_START_TIME_MODIFIED)
		swcode += 1; /*calculate end time*/;
 }
```